src/StatsdClient/RandomGenerator.cs

```diff
@@ -4,15 +4,26 @@ namespace StatsdClient
 {
     public class RandomGenerator : IRandomGenerator
     {
-        readonly Random _random;
+        [ThreadStatic]
+        static Random _random;
+
+        private static Random Random
+        {
+            get
+            {
+                var random = _random;
+                if (random != null) return random;
+                return _random = new Random(Guid.NewGuid().GetHashCode());

```

Comment:
You can't initiate _random in the static constructor because the static constructor won't necessarily run on every thread. Therefore the thread that ran the static constructor will have an initialized _random and all other threads will have a null _random.

The reason I seed with `Guid.NewGuid().GetHashCode()` rather than the default  is that the default constructor uses the current time as a seed. If all `_random` instances are initialized around the same time then they will get the same seed.